crates/biome_js_analyze/src/lint/correctness/no_undeclared_dependencies.rs

```diff
@@ -34,32 +38,199 @@ declare_lint_rule! {
+                Self::FilesGlob(b) => a
+                    .iter()
+                    .zip(b.iter())
+                    .all(|(a, b)| a.to_string() == b.to_string()),
+                _ => false,
+            },
+        }
+    }
+}
+
+impl Eq for DependencyAvailability {}
+
+impl Deserializable for DependencyAvailability {
+    fn deserialize(
+        value: &impl biome_deserialize::DeserializableValue,
+        name: &str,
+        diagnostics: &mut Vec<biome_deserialize::DeserializationDiagnostic>,
+    ) -> Option<Self> {
+        Some(if value.visitable_type()? == DeserializableType::Bool {
+            if bool::deserialize(value, name, diagnostics)? {
+                Self::Available
+            } else {
+                Self::Unavailable
+            }
+        } else {
+            Self::FilesGlob(Deserializable::deserialize(value, name, diagnostics)?)
+        })
+    }
+}
+
+#[cfg(feature = "schemars")]
+impl schemars::JsonSchema for DependencyAvailability {
+    fn schema_name() -> String {
+        "DependencyAvailability".to_owned()
+    }
+
+    fn json_schema(_gen: &mut schemars::gen::SchemaGenerator) -> schemars::schema::Schema {
+        use schemars::schema::*;
+
+        Schema::Object(SchemaObject {
+            subschemas: Some(Box::new(SubschemaValidation {
+                one_of: Some(vec![
+                    Schema::Object(SchemaObject {
+                        instance_type: Some(InstanceType::Boolean.into()),
+                        metadata: Some(Box::new(Metadata {
+                            description: Some("This type of dependency will be always available or unavailable.".to_owned()),
+                            ..Default::default()
+                        })),
+                        ..Default::default()
+                    }),
+                    Schema::Object(SchemaObject {
+                        instance_type: Some(InstanceType::Array.into()),
+                        array: Some(Box::new(ArrayValidation {
+                            items: Some(SingleOrVec::Single(Box::new(Schema::Object(SchemaObject {
+                                instance_type: Some(InstanceType::String.into()),
+                                ..Default::default()
+                            })))),
+                            min_items: Some(1),
+                            ..Default::default()
+                        })),
+                        metadata: Some(Box::new(Metadata {
+                            description: Some("This type of dependency will be available only if the linted file matches any of the globs.".to_owned()),
+                            ..Default::default()
+                        })),
+                        ..Default::default()
+                    })
+                ]),
+                ..Default::default()
+            })),
+            ..Default::default()
+        })
+    }
+}
+
+impl DependencyAvailability {
+    fn is_available(&self, path: &str) -> bool {
```

Comment:
We could use `Path` instead of `str`:

```suggestion
    fn is_available(&self, path: &Path) -> bool {
```

Reply:
Fixed in [a74e26b](https://github.com/biomejs/biome/pull/4376/commits/a74e26b252e1f2c87028eaafa3fce6edccb74b22)